lib/sisimai/bite/email/sendmail.rb

```diff
@@ -0,0 +1,290 @@
+module Sisimai::Bite::Email
+  module Sendmail
+    # Sisimai::Bite::Email::Sendmail parses a bounce email which created by
+    # v8 Sendmail. Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Sendmail.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/\AMail Delivery Subsystem/,
+        :subject => %r/(?:see transcript for details\z|\AWarning: )/,
+      }
+      # Error text regular expressions which defined in sendmail/savemail.c
+      #   savemail.c:1040|if (printheader && !putline("   ----- Transcript of session follows -----\n",
+      #   savemail.c:1041|          mci))
+      #   savemail.c:1042|  goto writeerr;
+      #
+      Re1 = {
+        :begin   => %r/\A[ \t]+[-]+ Transcript of session follows [-]+\z/,
+        :error   => %r/\A[.]+ while talking to .+[:]\z/,
+        :rfc822  => %r{\AContent-Type:[ ]*(?:message/rfc822|text/rfc822-headers)\z},
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'V8Sendmail: /usr/sbin/sendmail'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Sendmail
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        unless mhead['subject'] =~ /\A[ \t]*Fwd?:/i
+          # Fwd: Returned mail: see transcript for details
+          # Do not execute this code if the bounce mail is a forwarded message.
+          return nil unless mhead['from'] =~ Re0[:from]
+        end
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [170.8/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->